### DISCUSSION TITLE
showWholeGraph = 'when-maximised' instead of 'true'?

### DISCUSSION THREAD

**Discussion by 9mm** [jesseduffield/lazygit#3014] [General]
I really love the combined git log graph where it shows a visual of the branches and commits:

```
    showGraph: when-maximised
    showWholeGraph: true
```

The problem is that when the commits pane is small, i want to see neither a graph, nor all branch commits in one pane

but when its maximized, i want both

currently the showGraph setting works great when maximized, but i when its maximized thats when i want to see all commits together (given the fact that the visual helps you differentiate between them)

Right now i have to set `showWholeGraph` to false becuase its just way too confusing to see mixed in commits with no graph in a tiny window

Is there a way to get

`showWholeGraph` to have the same options as `showGraph`? (ie... `when-maximized`, `always` etc)

**Comment by mbhutton** [comment]
I'm similar in wanting just the current branch when the panel is minimised, but when it's maximised I sometimes want to show all branches, and sometimes just want just the current branch. 

Would a toggle also make sense, eg `ctrl-a` to toggle showing all branches vs not? To override the configured value for duration of the session. (Keybindings for `a` and `A` are already taken.)

Presumably `true` and `false` would need to keep being supported for backwards compatibility, as synonyms for `always` and `never`.
